lib/keyfile/keyfile.c

```diff
@@ -372,27 +372,31 @@ keyfile_read(const char * filename, uint64_t * machinenum, int keys)
 	struct stat sb;
 	uint8_t * keybuf;
 	FILE * f;
+	size_t keyfilelen;
 
 	/* Stat the file. */
 	if (stat(filename, &sb)) {
 		warnp("stat(%s)", filename);
 		goto err0;
 	}
 
-	/* Allocate memory. */
-	if ((sb.st_size == 0) || (sb.st_size > 1000000)) {
+	/* Validate keyfile size. */
+	keyfilelen = (size_t)(sb.st_size);
```

Comment:
Do you want the parentheses around `(sb.st_size)` ?  My mental model of your normal style suggested that they shouldn't be there, but perhaps this clarifies that the `(size_t)` refers to the `.st_size` portion of the structure, not the `sb` structure itself?